book/concepts.md

```diff
@@ -1,6 +1,131 @@
 # Concepts
+Hydroflow is different from other distributed systems infrastructure, so you probably have questions, like:
+What is Hydroflow? How does it work? What is special about it, and what are the key concepts to understand?
+
+This chapter covers those basic questions. We start simple describing Hydroflow, and build up to an understanding of 
+what makes Hydroflow uniquely powerful.
+
+But in case you want a preview of the Hydroflow goodies, here are the main themes:
+1. **Distributed Correctness**: Hydroflow's type system can prevent distributed system bugs at compile time. (One example: will your code
+produce the same results if you deploy on a single machine or replicate on a distributed cluster of machines?)
+2. **Local Efficiency**: Hydroflow compiles your dataflow programs into efficient, low-latency, single-threaded executables.
+
+More on these topics below.
+
+## Hydroflow: A Runtime Library and a DSL in Rust
+Let's skip the fanfare and keep things simple: what *is* Hydroflow?
+
+Hydroflow is a library that can be used in any Rust program. It includes two main components:
+
+1. A runtime library that executes low-latency, reactive dataflow programs written in Rust. (The *core API*.)
+2. A domain-specific language (DSL) for specifying dataflow programs. (The Hydroflow *surface syntax*.) 
+
+Hydroflow's surface syntax must be embedded in a Rust program; the Rust compiler takes that Hydroflow syntax and 
+compiles it into an efficient binary executable we call a Hydroflow *spinner*.
+
+In typical usage, a developer writes a spinner as a single-threaded Rust program that is mostly composed of 
+Hydroflow surface syntax. Each spinner is typically responsible for a single 
+"node" (a machine, or a core) in a distributed system composed of many such spinners,
```

Comment:
I know "spinner" is just a placeholder. But is it not just a node? At least 1:1 with the concept of a node